services/filter_file.rb

```diff
@@ -0,0 +1,62 @@
+require_relative '../translation'
+
+class FilterFile
+  include ::Translation
+
+  def call(wait_time = 1)
+    translate(:enter_path)
+    input_file = gets.chomp
+
+    unless File.exist?(input_file)
+      translate(:file_not_found)
+      return
+    end
+
+    translate(:enter_filtering_words)
+    target_words = gets.chomp.split(',').map { _1.strip}
+
+    total_rows = CSV.read(input_file).size
+
+    progress_bar = ProgressBar.create(
+      title: I18n.t(:filtering),
+      total: total_rows,
+      format: "%t: |%B| %p%% %e"
+    )
+
+    modified_rows = []
+    mod_words_count = 0
+
+    CSV.foreach(input_file) do |row|
+      row.map! do |value|
+        words = value.split(" ")
+        words.map! do |word|
+          if target_words.include?(word)
+            mod_words_count += 1
+            '*'
+          else
+            word
+          end
+        end
```

Comment:
This method is really long and does a lot, which violates [SRP](https://medium.com/@asvetly/solid-ruby-ad046727ec26). You can use a [bunch](https://refactoring.com/catalog/extractFunction.html) of refactorings here.
```ruby
# up to you how to name and how many extracted functions you'll have
def call
  get_file_path_input
  check_file_presence
  get_list_of_words
  initiate_progressbar
  filter_file
  create_file_copy
  show_result
end

private

def get_file_path_input
  # something here
end
```

Reply:
Split it into multiple methods